File: hrms-frontend/src/services/projectService.js
// src/services/projectService.js
import api from '../api/axiosConfig';

const projectService = {
  getAll: () => api.get('/projects'),
  getActive: () => api.get('/projects/active'),
  getById: (id) => api.get(`/projects/${id}`),
  getByEmployee: (empId) => api.get(`/projects/employee/${empId}`),
  create: (data) => api.post('/projects', data),
  update: (id, data) => api.put(`/projects/${id}`, data),
  delete: (id) => api.delete(`/projects/${id}`),
  updateStatus: (id, status) =>
    api.patch(`/projects/${id}/status?status=${status}`),

  // Members
  addMember: (projectId, data) =>
    api.post(`/projects/${projectId}/members`, data),
  removeMember: (projectId, empId) =>
    api.delete(`/projects/${projectId}/members/${empId}`),
  getMembers: (projectId) => api.get(`/projects/${projectId}/members`),

  // Tasks
  createTask: (projectId, data) =>
    api.post(`/projects/${projectId}/tasks`, data),
  updateTask: (taskId, data) => api.put(`/projects/tasks/${taskId}`, data),
  updateTaskStatus: (taskId, status) =>
    api.patch(`/projects/tasks/${taskId}/status?status=${status}`),
  getTasks: (projectId) => api.get(`/projects/${projectId}/tasks`),
  getMyTasks: (empId) => api.get(`/projects/tasks/my/${empId}`),
  deleteTask: (taskId) => api.delete(`/projects/tasks/${taskId}`),

  getStatistics: () => api.get('/projects/statistics'),
};

export default projectService;

File: hrms-backend/src/main/java/com/hrms/service/ProjectService.java

//service/ProjectService.java
package com.hrms.service;

import com.hrms.dto.response.ProjectResponse;
import com.hrms.model.Project;
import com.hrms.model.ProjectMember;
import com.hrms.model.ProjectTask;

import java.util.List;
import java.util.Map;

public interface ProjectService {
 // Projects
 ProjectResponse createProject(Project project);
 ProjectResponse updateProject(Long id, Project project);
 ProjectResponse getProject(Long id);
 List<ProjectResponse> getAllProjects();
 List<ProjectResponse> getActiveProjects();
 List<ProjectResponse> getProjectsByEmployee(Long employeeId);
 void deleteProject(Long id);
 ProjectResponse updateStatus(Long id, String status);

 // Members
 ProjectMember addMember(Long projectId, Long employeeId, String role);
 void removeMember(Long projectId, Long employeeId);
 List<ProjectMember> getMembers(Long projectId);

 // Tasks
 ProjectTask createTask(Long projectId, ProjectTask task);
 ProjectTask updateTask(Long taskId, ProjectTask task);
 ProjectTask updateTaskStatus(Long taskId, String status);
 List<ProjectTask> getTasksByProject(Long projectId);
 List<ProjectTask> getTasksByEmployee(Long employeeId);
 void deleteTask(Long taskId);

 // Statistics
 Map<String, Object> getStatistics();
}